src/lib/__tests__/InstantSearch-test.js

```diff
@@ -1603,12 +1615,51 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/instantsear
     expect(searchClient.search).toHaveBeenCalledTimes(2);
   });
 
-  test('notifies all middleware', () => {
+  test('notifies all middleware', async () => {
+    const searchClient = createSearchClient();
+    const search = new InstantSearch({
+      indexName: 'indexName',
+      searchClient,
+    });
+    const onMiddlewareStateChange = jest.fn();
+    const middleware = () => {
+      return {
+        subscribe() {},
+        unsubscribe() {},
+        onStateChange: onMiddlewareStateChange,
+      };
+    };
+
+    search.EXPERIMENTAL_use(middleware);
```

Comment:
```suggestion
    search.use(middleware);
```

You can just use `use` now :)